go/pkg/pwcompose/compose.go

```diff
@@ -235,14 +238,44 @@ func Up(
 		}
 	}
 
+	// create instances
+	logger.Debug("docker-compose", zap.String("action", "up --no-start"))
+	cmd := exec.Command("docker-compose", "-f", tmpPreparedComposePath, "up", "--no-start")
+	cmd.Stdout = os.Stderr
+	cmd.Stderr = os.Stderr
+	err = cmd.Run()
+	if err != nil {
+		logger.Error("Error detected while creating containers, it's probably due to a conflict with previously created containers that share the same name. You should retry with --force-recreate flag")
+		return errcode.ErrComposeRunCreate.Wrap(err)
+	}
+
+	// copy pathwar binary inside all containers
+	pwInfo, err := GetPathwarInfo(ctx, cli)
+	if err != nil {
+		return errcode.ErrComposeGetPathwarInfo.Wrap(err)
+	}
+
+	for _, service := range pwInfo.RunningInstances {
+		if challengeID == fmt.Sprintf("%s@%s", service.Labels[challengeNameLabel], service.Labels[challengeVersionLabel]) {
```

Comment:
You can create an helper for this:

```go
fmt.Sprintf("%s@%s", service.Labels[challengeNameLabel], service.Labels[challengeVersionLabel]
```

in order to call this:

```go
service.ChallengeID()
```

Reply:
Something like this in config.go ?

`type service struct {
	ContainerName                             string            `yaml:"container_name,omitempty"`
	Image                                     string            `yaml:",omitempty"`
	Networks, Ports, Expose, Volumes, Command []string          `yaml:",omitempty"`
	VolumesFrom                               []string          `yaml:"volumes_from,omitempty"`
	DependsOn                                 []string          `yaml:"depends_on,omitempty"`
	CapAdd                                    []string          `yaml:"cap_add,omitempty"`
	Build                                     string            `yaml:",omitempty"`
	Restart                                   string            `yaml:",omitempty"`
	Environment                               map[string]string `yaml:",omitempty"`
	Labels                                    map[string]string `yaml:"labels,omitempty"`
}

func (s service) ChallengeID() string {
	return fmt.Sprintf("%s@%s", s.Labels[challengeNameLabel], s.Labels[challengeVersionLabel])
}`

Reply:
Done